meson.build

```diff
@@ -290,6 +290,7 @@ test_flags = ['-Wdisabled-optimization',
               '-Wno-sign-compare',
               '-Wno-switch',
               '-Wno-unused-parameter',
+              '-Woverflow',
```

Comment:
Move this up to alongside `-W`, here are `-Wno-`, also keep it sorted alphabetically withing that group.